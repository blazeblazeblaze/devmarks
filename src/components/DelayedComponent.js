import React from "react";
import PropTypes from "prop-types";

const propTypes = {
  isReady: PropTypes.bool.isRequired,
  fallbackComponent: PropTypes.element.isRequired,
  children: PropTypes.element.isRequired
};

const DelayedComponent = ({ isReady, fallbackComponent, children }) => (
  <div>{isReady ? children : fallbackComponent}</div>
);

DelayedComponent.propTypes = propTypes;

export default DelayedComponent;
