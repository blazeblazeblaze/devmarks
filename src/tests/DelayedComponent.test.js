import React from "react";
import DelayedComponent from "../components/DelayedComponent";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<DelayedComponent />", () => {
  const MainComponent = () => <div>main component</div>;
  const Placeholder = () => <div>placeholder component</div>;

  describe("when isReady is true", () => {
    it("renders its children", () => {
      const tree = shallow(
        <DelayedComponent fallbackComponent={<Placeholder />} isReady={true}>
          <MainComponent />
        </DelayedComponent>
      );

      expect(toJson(tree)).toMatchSnapshot();
    });
  });

  describe("when isReady is false", () => {
    it("renders placeholder component", () => {
      const tree = shallow(
        <DelayedComponent fallbackComponent={<Placeholder />} isReady={false}>
          <MainComponent />
        </DelayedComponent>
      );

      expect(toJson(tree)).toMatchSnapshot();
    });
  });
});
