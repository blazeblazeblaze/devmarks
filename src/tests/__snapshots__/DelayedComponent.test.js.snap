// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DelayedComponent /> when isReady is false renders placeholder component 1`] = `
<div>
  <Placeholder />
</div>
`;

exports[`<DelayedComponent /> when isReady is true renders its children 1`] = `
<div>
  <MainComponent />
</div>
`;
